e2e-test/src/tests/delete.test.js

```diff
@@ -0,0 +1,66 @@
+import { vBtn } from 'jest-puppeteer-vuetify';
+import { CLIENT_URL } from '../util';
+
+
+describe('delete dataset', () => {
+  it('upload one dataset and then delete it', async () => {
+    await Promise.all([
+      page.goto(CLIENT_URL),
+      page.waitForNavigation({ waitUntil: 'networkidle0' }),
+    ]);
+    await expect(page).toClickXPath(vBtn('Your Datasets'));
+
+    const fileInputElementHandle = await page.waitForXPath("//input[@type='file']"); // $0.value == "" is true
+    await fileInputElementHandle.uploadFile('../e2e-test/data/Cancer chemo muscle (NMR).csv');
+    await page.waitForXPath("//div[@class='v-list__tile__title'][contains(text(),'Cancer_chemo_muscle_NMR.csv ')]");
+    await page.waitForXPath("//div[@class='v-list__tile__title'][contains(text(),'Cancer_chemo_muscle_NMR.csv ')]//following::span[contains(text(),'Dataset processed with 1 validation failures')]");
+
+    // click on 'X'
+    await page.waitForXPath("//div[@class='v-btn__content']//following::i[@class='v-icon mdi mdi-close theme--light']");
+    await expect(page).toClickXPath(vBtn("//i[@class='v-icon mdi mdi-close theme--light']"));
+
+    // make sure it is deleting one dataset
+    await page.waitForXPath("//div[@class='v-card__title headline'][contains(text(),'Really delete 1 dataset?')]");
+    await expect(page).toClickXPath(vBtn('Delete'));
+
+    // make sure it is deleted
+    await expect(page).not.toContainXPath("//div[@class='v-list__tile__title'][contains(text(),'Cancer_chemo_muscle_NMR.csv ')]");
```

Comment:
I'm curious what happens if you run the upload test and then run this test. The upload test should leave behind the files that it uploaded, including this one. When this test runs it uploads the file again, but only deletes one, so after the deletion happens the file should still be on the page. Maybe this assertion is not doing what it is supposed to be doing?

The solution to this would be to use different files for different tests. For now it would be easiest to just have unique data files for every test. Eventually we might consider writing a helper function that takes a filename, appends some unique identifier to it, copies the original data file to the new longer file name, then uploads that. Seems like overkill for the tests we have now though.

Reply:
I did not realize that the files are left behind that is probably why sometimes the CI tests are failing. Because there are more than 3 datasets already. I will try to use unique datasets for different tests. So far I was able to download the three from https://viime.org/#/ are there any more datasets that I can use? Where can I find more?

Reply:
The easiest thing would be to just take those datasets and rename them. The data will be the same, but they will be considered unique by VIIME.

If you ever do need different data to test a feature or something, you can try the [metaboanalyst samples](https://www.metaboanalyst.ca/MetaboAnalyst/upload/StatUploadView.xhtml). They're harder to download but there are more of them.